azuredevops/client_repository_commit.go

```diff
@@ -112,12 +112,13 @@ func (c *CommitClient) Create(ctx context.Context, branch string, message string
 	// get latest commit from branch
 
 	commitsList, err := c.ListPage(ctx, branch, 1, 0)
-	if err != nil {
-		return nil, err
+	latestCommitTreeSHA := ""
```

Comment:
you will never get an "404" because `c.g.GetCommits` discards the http.Response.StatusCode, It returns an error only if it cannot read the response body, client policy (such as CheckRedirect), or failure to speak HTTP.

So if it doesn't find it, you just get a `nil` `commitsList`.